xcm/xcm-builder/src/barriers.rs

```diff
@@ -241,12 +248,12 @@ impl<T: Contains<MultiLocation>> ShouldExecute for AllowExplicitUnpaidExecutionF
 			origin, instructions, max_weight, _weight_credit,
 		);
 		ensure!(T::contains(origin), ());
-		match instructions.first() {
-			Some(UnpaidExecution { weight_limit: Limited(m), .. }) if m.all_gte(max_weight) =>
-				Ok(()),
-			Some(UnpaidExecution { weight_limit: Unlimited, .. }) => Ok(()),
+		instructions.matcher().match_next_inst(|inst| match inst {
+			UnpaidExecution { weight_limit: Limited(m), .. } if m.all_gte(max_weight) => Ok(()),
+			UnpaidExecution { weight_limit: Unlimited, .. } => Ok(()),
 			_ => Err(()),
-		}
+		})?;
+		Ok(())
```

Comment:
```suggestion
		})
```

Reply:
```suggestion
		})
		.map(|_| ())
```
But this is why I opted for the `Ok(())` at the end instead.